src/BaGet/Entities/PostgreSqlContext.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+
+namespace BaGet.Entities
+{
+    public class PostgreSqlContext : AbstractContext<PostgreSqlContext>
+    {
+        /// <summary>
+        /// The PostgreSql error code for when a unique constraint is violated.
+        /// </summary>
+        private const string UniqueConstraintViolationErrorCode = "23505";
+
+        public PostgreSqlContext(DbContextOptions<PostgreSqlContext> options)
+            : base(options)
+        {
+        }
+
+        public override bool IsUniqueConstraintViolationException(DbUpdateException exception)
+        {
+            return exception.InnerException is PostgresException postgresException &&
+                   postgresException.SqlState.Contains(UniqueConstraintViolationErrorCode);
+        }
+    }
```

Comment:
Please override `OnModelCreating` and mark the `Package.Id` and `PackageDependency.Id` properties as case insensitive. See @Mizipzor's work here: https://github.com/loic-sharma/BaGet/pull/163/files#diff-7dd1af1d5bde323eb9b01ee2fc2671c4R27